src/main/java/com/nike/cerberus/server/config/guice/CmsGuiceModule.java

```diff
@@ -287,4 +298,56 @@ public StaticAssetManager dashboardStaticAssetManager() {
         int maxDepthOfFileTraversal = 2;
         return new StaticAssetManager(DASHBOARD_DIRECTORY_RELATIVE_PATH, maxDepthOfFileTraversal);
     }
+
+    @Provides
+    @Singleton
+    @Named("encryptCryptoMaterialsManager")
+    public CryptoMaterialsManager encryptCryptoMaterialsManager(@Named("cms.encryption.cmk.arns") String cmkArns,
+                                                                @Named("cms.encryption.encrypt.cache.maxSize") int maxSize,
+                                                                @Named("cms.encryption.encrypt.cache.maxAge") int maxAge,
+                                                                @Named("cms.encryption.encrypt.cache.messageUseLimit") int messageUseLimit,
+                                                                Region currentRegion) {
+        logger.info("Initializing encryptCryptoMaterialsManager with CMK: {}, maxSize: {}, maxAge: {}, " +
+                "messageUseLimit: {}", cmkArns, maxSize, maxAge, messageUseLimit);
+        MasterKeyProvider<KmsMasterKey> keyProvider = initializeKeyProvider(cmkArns, currentRegion);
+        CryptoMaterialsCache cache = new LocalCryptoMaterialsCache(maxSize);
+        CryptoMaterialsManager cachingCmm =
+                CachingCryptoMaterialsManager.newBuilder().withMasterKeyProvider(keyProvider)
+                        .withCache(cache)
+                        .withMaxAge(maxAge, TimeUnit.SECONDS)
+                        .withMessageUseLimit(messageUseLimit)
```

Comment:
You don't really need any caching or re-use on write since writes are so infrequent.  If you are going to spend memory on a cache it seems better to spend zero on write and spend all of it on read.

Reply:
We generally agree with you, but with we figured we would like a tiny cache here just incase a user goes off the rails and does something wrong so that we don't blow through our KMS API limit and inadvertently get DDOS'd by a user. 

With the mandated move away from AWS WAF to our internal WAF we are going to lose our current rate limiter unless we re-engineer it to be a 1st class thing in the app.

Under normal usages, we won't really end up using the cache because they are so infrequent, so it's really there just in-case some sort of burst traffic happens due to a user doing something wrong.